### DISCUSSION TITLE
[Q&A] how can i run my self hosted llms using opanai compatible api but with no key

### DISCUSSION THREAD

**Discussion by Ghaithkhelifi** [lobehub/lobehub#959] [Development Question] [answered]
how can i run my self-hosted llms using opanai compatible API but with no key 
all the time when i past my api url i always get the required key 
i use openllm for api or localai 
please tell me how to use my own llm

**Comment by arvinxx** [comment, marked as answer]
you can add a random key like "abc".

and set the OpenAI Proxy URL to your local llm url.

**Comment by tidianestano-star** [comment]
If you don't want to maintain the local routing and translation layer yourself (which gets tedious when new models drop), look into a hosted unified gateway like 4sapi.com. They already mapped virtually every model family into the standard /v1/chat/completions schema. You just use one master key and can flexibly switch between western models and open-source models without changing a single line of your configuration.
